src/js/common/data_graphic.js

```diff
@@ -11,126 +11,155 @@ MG.deprecations = {
 MG.globals.link = false;
 MG.globals.version = "1.1";
 
+MG.options = { // <name>: [<defaultValue>, <availableType>]
+  x_axis_type: [null, ['categorical']], // TO BE INTRODUCED IN 2.10
+  y_axis_type: [null, ['categorical']], // TO BE INTRODUCED IN 2.10
+  y_padding_percentage: [0.05, 'number'],                 // for categorical scales
+  y_outer_padding_percentage: [0.1, 'number'],            // for categorical scales
+  ygroup_padding_percentage: [0.25, 'number'],            // for categorical scales
+  ygroup_outer_padding_percentage: [0, 'number'],         // for categorical scales
+  x_padding_percentage: [0.05, 'number'],                 // for categorical scales
+  x_outer_padding_percentage: [0.1, 'number'],            // for categorical scales
+  xgroup_padding_percentage: [0.25, 'number'],            // for categorical scales
+  xgroup_outer_padding_percentage: [0, 'number'],         // for categorical scales
```

Comment:
@wlach Could you take a look at them? I am not very clear what is their effect.

Reply:
Just had a look, pretty confusing. :/ https://codepen.io/wlach/pen/LBvwLw

You can find where these are defined/used by doing a grep for the part that isn't dependent on the x/y. e.g. `git grep out_padding_percentage`.

As the comments implies, it appears these are only used for categorical scales. When it comes time to document them, we can just use a generic comment like "Controls padding for categorical scales". Or maybe we can just hide these options from the user, since I don't know if this is even something we should be exposing for now.